### ISSUE TITLE
Sporadic test failures using during visit then wait

### ISSUE THREAD

**Issue by daniel-seitz** [cypress-io/cypress#18524]
### Current behavior

I have sporadic failures in my tests (not always reproducable) when writing

cy.intercept(...).as('request-finished')
cy.visit('/');
cy.wait('@request-finished');

It seems like the timeout (5 sec in my case) already starts to run before the page is fully initialized. See the screenshot - which wouldn't be much of a problem if the interceptor would see the corresponding XHR request.

![image](https://user-images.githubusercontent.com/12377678/137700974-1b006cbf-7bb2-48fd-97e8-816d60d76815.png)

After this little blue line expires then the error occurs although the XHR is there as you can see in the image above:

![image](https://user-images.githubusercontent.com/12377678/137706357-fa8e6af2-d25f-4cbe-8c0c-4290f5a43ecd.png)

**In conclusion, the intercept does sometimes not detect the XHR request.**

Any hints on further debugging are appreciated. (console.log statements / cy.log happen to fast)

I know that increasing the timeout would be possible however I don't think that this is a good solution.

### Desired behavior

Cypress should wait

### Test code to reproduce

cy.intercept(...).as('request-finished')
cy.visit('/');
cy.wait('@request-finished');

### Cypress Version

8.6.0

**Comment by esetnik** [comment]
I am having the same issue using `cy.intercept()` before `cy.visit()`. I'm getting sporadic failures of the intercept to catch the XHR request.

**Comment by liebsterkamerad** [comment]
We already had this topic [3427](https://github.com/cypress-io/cypress/issues/3427?utm_source=pocket_mylist) and it looks like the problem persists.

It works without any flaws locally but then in CI pipeline cy.wait() sometimes fails even though the response was actually OK. And it actually worked 7 times before during the same test.

Here's a screenshot of the pipeline failure, it looks like wait starts later that the request occurs.
![image](https://user-images.githubusercontent.com/12257375/139846461-462f18bd-f25f-41aa-81a3-c96f8d88f211.png)
